canvasapi/course.py

```diff
@@ -2347,6 +2349,51 @@ def list_blueprint_subscriptions(self, **kwargs):
             kwargs=combine_kwargs(**kwargs)
         )
 
+    def get_grading_periods(self, **kwargs):
+        """
+        Return a list of grading periods for the associated course.
+
+        :calls: `GET /api/v1/courses/:course_id/grading_periods\
+        <https://canvas.instructure.com/doc/api/grading_periods.html#method.grading_periods.index>`_
+
+        :rtype: :class:`canvasapi.paginated_list.PaginatedList` of
+            :class:`canvasapi.grading_period.GradingPeriod`
+        """
+
+        return PaginatedList(
+            GradingPeriod,
+            self._requester,
+            'GET',
+            'courses/{}/grading_periods'.format(
+                self.id
+            ),
+            {'course_id': self.id},
+            _root="grading_periods",
+            kwargs=combine_kwargs(**kwargs)
+        )
+
+    def get_grading_period(self, grading_period_id, **kwargs):
+        """
+        Return a single grading period for the associated course and id.
+
+        :calls: `GET /api/v1/courses/:course_id/grading_periods/:id\
+        <https://canvas.instructure.com/doc/api/grading_periods.html#method.grading_periods.index>`_
+        :param grading_period_id: The ID of the rubric.
+        :type grading_period_id: int
+
+        :rtype: :class:`canvasapi.grading_period.GradingPeriod`
+        """
+
+        response = self._requester.request(
+            'GET',
+            'courses/{}/grading_periods/{}'.format(
+                self.id, grading_period_id
+            ),
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+        return GradingPeriod(self._requester, response.json()['grading_periods'][0])
```

Comment:
I see you pass `{'course_id': self.id}` to `get_grading_periods`, implying that `course_id` is not returned by Canvas. If this is the case, you may need to manually add `course_id` to the object here as well. Check all other methods for the same thing.